src/stageros.cpp

```diff
@@ -797,13 +800,13 @@ main(int argc, char** argv)
     if(sn.SubscribeModels() != 0)
         exit(-1);
 
-    boost::thread t = boost::thread(boost::bind(&ros::spin));
+//    std::thread t = std::thread(rclcpp::spin);
 
     sn.world->Start();
 
     Stg::World::Run();
     
-    t.join();
+//    t.join();
```

Comment:
Should there not be a spin here somewhere?

Reply:
Indeed. Brought the extra thread back with a call to `rclcpp::spin` and the `Node::SharedPtr` from the `StageNode` class.